File: virtuelle_physik/Exercise 1/Yeast.m
%% yeast
V = 1;
s0 = 2;
p0 = 1e-6;

Cf = 50;
R = 10;
S = 15;
Tref = 300;

h = 0.01;
n = 1;

p(n) = p0;
s(n) = s0;

t = 0;

T = 310;

while t<=10
    p(n+1) = p(n)+p(n)*((R+2*S)*s(n)-2*S*s0)*h;
    s(n+1) = s(n)-(s(n)*p(n)*Cf*(T/Tref)/V)*h;
    t = t+h;
    n = n+1;
end

a = (s0 - s);

plot(a,'r')
hold on
grid on
plot(s)

figure
grid on
plot(p)

File: virtuelle_physik/Exercise 1/yeast.py
#!/usr/bin/env python3


#Setting the parameters
V = 1  #volume of fermentation vessel
s0 = 0.2 #inital concentration of sugar 
p0 = 1e-6 #initial population of yeast [m3]

C_f = 50 #feeding Coefficient [1/day]
R = 10 #reproductivity [1/day]
S = 15 #sensitivity w.r.t. alcohol [1/day]
T_ref = 300 #reference temperature [K]

h = 0.01 #time-step of forward Euler integration


#Setting the initial values
p = p0
s = s0
a = s0 - s;
time = 0


#Setting the input-value
T = 310


#perform time-integration
while time < 10:
    dp_dt = p*((R+S)*s - S*s0)
    ds_dt = -s*p*C_f*(T/T_ref)*1/V
    p += h*dp_dt
    s += h*ds_dt
    a = s0-s
    time += h
    print(time,"\t",p,"\t",s,"\t",a)
